pkg/operator/ceph/csi/secrets.go

```diff
@@ -212,9 +259,175 @@ func CreateCSISecrets(context *clusterd.Context, clusterInfo *client.ClusterInfo
 		return errors.Wrap(err, "failed to create kubernetes csi secret")
 	}
 
+	err = updateCephStatusWithCephxStatus(context, clusterInfo, clusterNamespaced, currentKeyCount)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// updateCephStatusWithCephxStatus updates the cephCluster status with CSI cephxStatus
+func updateCephStatusWithCephxStatus(context *clusterd.Context, clusterInfo *client.ClusterInfo, namespacedName types.NamespacedName, currentKeyCount int) error {
+	logger.Info("updating cephStatus with CSI cephxStatus")
+	err := retry.OnError(
+		retry.DefaultRetry,
+		func(err error) bool {
+			return err != nil
+		},
+		func() error {
+			cephCluster, err := getCephcluster(context, clusterInfo, namespacedName)
+			if err != nil {
+				return err
+			}
+
+			keyRotationEnabled, err := keyring.ShouldRotateCephxKeys(cephCluster.Spec.Security.CephX.CSI.CephxConfig, clusterInfo.CephVersion, clusterInfo.CephVersion, cephv1.CephxStatus{})
+			if err != nil {
+				return errors.Wrap(err, "failed to determine if cephx keys should be rotated")
+			}
+
+			cephxStatus := keyring.UpdatedCephxStatus(keyRotationEnabled, cephCluster.Spec.Security.CephX.Daemon, clusterInfo.CephVersion, *cephCluster.Status.Cephx.CSI.CephxStatus)
+			cephCluster.Status.Cephx.CSI.CephxStatus = &cephxStatus
+			cephCluster.Status.Cephx.CSI.PriorKeyCount = uint8(currentKeyCount) //nolint:gosec // disable G115
```

Comment:
Let's calculate the updated cephx status in the caller instead and then pass it to this method. RGW and CephClient are using this pattern, and keeping consistency will be good